### ISSUE TITLE
Prow job stuck in triggered state due to invalid cluster alias

### ISSUE THREAD

**Issue by ixdy** [kubernetes/test-infra#21836]
We recently debugged a prow job which was stuck in the triggered state. It was impossible to tell what was wrong by looking at deck, but looking at plank logs, we eventually figured out what was wrong, due to messages like

```
{"component":"plank","error":"errors syncing: [error starting pod: unknown cluster alias \"foo\"]","file":"prow/plank/controller.go:167","func":"k8s.io/test-infra/prow/plank.(*Controller).Start","level":"error","msg":"Error syncing.","severity":"error"}
```

The configuration had been copied from a different prow instance which had the "foo" build cluster defined, but this particular prow instance did not have a "foo" build cluster.

**What would you like to be added**:
Two things come to mind:
1. It'd be helpful if the plank error message gave more details about the job/pod it failed to sync (I initially tried grepping through the log for the job name and job ID, neither of which matched)
2. It'd be even more helpful if the prow job status had been updated in some way with this error such that it could be exposed in deck.

**Why is this needed**:
User error in configuration was not able to be debugged by the user without admin access to the prow cluster.

One additional note: this prow instance is using a fairly old build (`v20200729-2a25cd5714`), so it's possible some behavior has changed since then. I have not tried reproducing this issue on a newer build.

**Comment by alvaroaleman** [comment]
/area prow
This opens up the larger question if we want to mark the job as failed or keep it in the cluster in the hope that someone will add a `foo` cluster and we will then be able to process it.

**Comment by alvaroaleman** [comment]
I tend to argue for marking as failed, as most prow jobs are kept fairly shortly anyways and ppl can just rerun the job if someone does indeed add the cluster. WDYT @chaodaiG ?

**Comment by chaodaiG** [comment]
> I tend to argue for marking as failed, as most prow jobs are kept fairly shortly anyways and ppl can just rerun the job if someone does indeed add the cluster. WDYT @chaodaiG ?

This makes sense to me, it's definitely annoying enough to see jobs hanging for several days then failed with this error message.

A little bit into implementation detail:

- I think it needs to retry for 5-10 minutes to cope with build cluster master upgrade
- This handled by plank sounds reasonable to me, WDYT?

**Comment by alvaroaleman** [comment]
> I think it needs to retry for 5-10 minutes to cope with build cluster master upgrade

No, this is not if it can not reach the cluster, it is if it doesn't have a client. What could however happen is that it doesn't have a client because it couldn't reach the cluster at time of clientcreation, which made discovery and thus clientcreation fail :upside_down_face: (if it does have the client but can not reach the cluster it will just retry)

> This handled by plank sounds reasonable to me, WDYT?
:+1:

**Comment by chaodaiG** [comment]
> > I think it needs to retry for 5-10 minutes to cope with build cluster master upgrade
> 
> No, this is not if it can not reach the cluster, it is if it doesn't have a client. What could however happen is that it doesn't have a client because it couldn't reach the cluster at time of clientcreation, which made discovery and thus clientcreation fail 🙃 (if it does have the client but can not reach the cluster it will just retry)

Ah. I see, and agreed. So yeah plank should just fail it
